clientconn.go

```diff
@@ -639,7 +638,7 @@ func (ac *addrConn) resetTransport(closeTransport bool) error {
 			t.Close()
 		}
 		sleepTime := ac.dopts.bs.backoff(retries)
-		timeout := minConnectTimeout
+		timeout := internal.MinConnectTimeout
```

Comment:
In the "internal" package, add a var:
`var TestingSetMinConnectTimeout func(time.Duration)`

Then in the clientconn.go, add

```
func init() {
	internal.TestingCloseConns = func(t time.Duration) {
		minConnectTimeout = t
	}
}
```

Then in the test, you can call
`internal.TestSetMinConnectTimeout(time.Second) `
to set it to 1 second.